### DISCUSSION TITLE
Container without h2 database?

### DISCUSSION THREAD

**Discussion by jbruenner** [DependencyTrack/dependency-track#1740] [Ideas]
Hi,

I'm introducing DT to our IT service group. They scan new containers. So they fell over the vulnerabilities of the
h2 database. Is it possible to maintain two versions of DT: a test version with h2 and a productive one without?

Thanks a lot
Jörg

**Comment by nscuro** [comment]
Building another container just so it doesn't include a specific component is not something we're interested in doing.

We should update the h2 version we ship instead. However, AFAIK h2 does not backport security fixes, and we'd have to update h2 from 1.4 to 2.0, which introduces breaking changes. So everyone's existing installation that uses h2 will break. 

@stevespringett, I know you looked into this b/c I've heard you mentioning the h2 situation on multiple occasions. Any strong feelings on the subject?

In any case, if you're not using h2, you're not affected by any of its vulnerabilities.

**Comment by msymons** [comment]
@jbruenner, exactly what vulnerabilities are you concerned about in h2?

#### sonatype-2020-1324?
This is not a vulnerability...  I reported this one to the h2 team and got a detailed (and very speedy) response explaining why it is not a vulnerability.  I have logged this with OSSI:   https://github.com/OSSIndex/vulns/issues/299

#### CVE-2018-14335?
Only 1.4.197 and older versions are affected.  I have updated [OSSI Bug 277](https://github.com/OSSIndex/vulns/issues/277) to reflect this

**Reply by jbruenner** [reply]
Hi all, 

thank you for your answers.

> Building another container just so it doesn't include a specific component is not something we're interested in doing.

Sure, it's not easy to maintain another container. At the other hand I don't like the idea of having unused components 
in production environments ...

> exactly what vulnerabilities are you concerned about in h2?

CVE-2021-42392, CVE-2022-23221 and CVE-2018-14335.
I can't read from the CVE notes if I'm affected or not. Did I missed any document telling me that?

> CVE-2018-14335?
> Only 1.4.197 and older versions are affected. I have updated https://github.com/OSSIndex/vulns/issues/277 to reflect this

It looks as it hasn't worked: DT 4.5 shows this bug with the SBoM I got from the 4.5 change log page.

> In any case, if you're not using h2, you're not affected by any of its vulnerabilities.

That sound good. I can make the production team feeling better with this answer.

The scanner mentioned among others the following critical vulnerabilities:
                package    vulnerability ID         installed version
debian	dpkg	CVE-2022-1664	1.20.9
debian	libc-bin	CVE-2021-33574	2.31-13
debian	libc-bin	CVE-2022-23218	2.31-13
debian	libc-bin	CVE-2022-23219	2.31-13
debian	libc6		CVE-2021-33574	2.31-13
debian	libc6		CVE-2022-23218	2.31-13
debian	libc6		CVE-2022-23219	2.31-13
debian	libssl1.1	CVE-2022-1292	1.1.1k-1
debian	openssl	CVE-2022-1292	1.1.1k-1

These components arn't mentioned in the SBom of DT 4.5. Do they scan nonsense? Is the SBoM of DT v4.5 incomplete?

Thank you for your help
Jörg

**Reply by jbruenner** [reply]
Hi,

I got an update from our production team: The scan engine was updated. Now we have only two vulnerabilities making trouble:
CVE-2022-1292 and CVE-2022-1664. Are they any problem to secure operation of DT?

Thank you for your help
Jörg

**Reply by nscuro** [reply]
> Are they any problem to secure operation of DT?

Not really, no. Both vulnerabilities require file system access by the attacker (the c_rehash script in OpenSSL that is affected by CVE-2022-1292 iterates over local files, not sure why they flagged the attack vector as network).

**Comment by jbruenner** [comment]
Great, thank you.

Jörg Brünner
Beauftragter für Informationssicherheit der SK
Neue Rufnummer: Sie erreichen mich ab sofort unter 0351.564.10220.
___________________________________________
SÄCHSISCHE STAATSKANZLEI | STATE CHANCELLERY OF SAXONY
Archivstr. 1 | 01097 Dresden | Postanschrift: 01095 Dresden
Tel.: +49 (0)351.564. 10220 |

***@***.******@***.***%7C> www.sachsen.de<http://www.sachsen.de/>

Der Empfang von elektronisch signierten und/oder verschlüsselten
elektronischen Dokumenten ist möglich. Die öffentlichen Schlüssel der
Sächsischen Staatskanzlei finden Sie unter https://www.sachsen.de/kontakt.html
***@***.***


Von: Niklas ***@***.***>
Gesendet: Montag, 27. Juni 2022 10:42
An: DependencyTrack/dependency-track ***@***.***>
Cc: Brünner, Jörg - SK ***@***.***>; Mention ***@***.***>
Betreff: Re: [DependencyTrack/dependency-track] Container without h2 database? (Discussion #1740)


Are they any problem to secure operation of DT?

Not really, no. Both vulnerabilities require file system access by the attacker (the c_rehash script in OpenSSL that is affected by CVE-2022-1292<https://github.com/advisories/GHSA-qjmp-vmxc-7p8r> iterates over local files, not sure why they flagged the attack vector as network).
